presto-product-tests/src/main/java/com/facebook/presto/tests/SqlCancelTests.java

```diff
@@ -0,0 +1,191 @@
+        closer = Closer.create();
+        closer.register(() -> executor.shutdownNow());
+        closer.register(queryCanceller);
+    }
+
+    @AfterTestWithContext
+    public void cleanUp()
+            throws IOException
+    {
+        closer.close();
+    }
+
+    @Test(groups = {CANCEL_QUERY}, timeOut = 60_000L)
+    public void cancelCreateTable()
+            throws Exception
+    {
+        String tableName = "cancel_createtable_" + nanoTime();
+        String sql = format("CREATE TABLE %s AS SELECT * FROM tpch.sf1.lineitem", tableName);
+
+        runAndCancelQuery(sql);
+        assertThat(() -> query("SELECT * from " + tableName))
+                .failsWithMessage(format("Table hive.default.%s does not exist", tableName));
+    }
+
+    @Test(groups = {CANCEL_QUERY}, timeOut = 60_000L)
+    public void cancelInsertInto()
+            throws Exception
+    {
+        String tableName = "cancel_insertinto_" + nanoTime();
+        query(format("CREATE TABLE %s (orderkey BIGINT, partkey BIGINT, shipinstruct VARCHAR(25)) ", tableName));
+        String sql = format("INSERT INTO %s SELECT orderkey, partkey, shipinstruct FROM tpch.sf1.lineitem", tableName);
+        runAndCancelQuery(sql);
+        assertThat(query("SELECT * from " + tableName)).hasNoRows();
+        query("DROP TABLE " + tableName);
+    }
+
+    @Test(groups = {CANCEL_QUERY}, timeOut = 60_000L)
+    public void cancelSelect()
+            throws Exception
+    {
+        runAndCancelQuery("SELECT * FROM tpch.sf1.lineitem AS cancel_select_" + nanoTime());
+    }
+
+    private void runAndCancelQuery(String sql)
+            throws Exception
+    {
+        Future<?> queryExecution = executor.submit(() -> assertThat(() -> query(sql)).failsWithMessage("Query was canceled"));
+
+        cancelQuery(sql);
+
+        try {
+            queryExecution.get(30, SECONDS);
+        }
+        catch (TimeoutException e) {
+            queryExecution.cancel(true);
+            throw e;
+        }
+    }
+
+    private void cancelQuery(String sql)
+            throws InterruptedException
+    {
+        long endTime = System.currentTimeMillis() + SECONDS.toMillis(30);
+        while (System.currentTimeMillis() < endTime) {
+            String findQuerySql = "SELECT query_id from system.runtime.queries WHERE query = '%s' and state = 'RUNNING' LIMIT 2";
+            QueryResult queryResult = query(format(findQuerySql, sql));
+            checkState(queryResult.getRowsCount() < 2, "Query is executed multiple times");
+            if (queryResult.getRowsCount() == 1) {
+                String queryId = (String) queryResult.row(0).get(0);
+                Response response = queryCanceller.cancel(queryId);
+                if (response.getStatusCode() != HttpStatus.OK.code()
+                        && response.getStatusCode() != HttpStatus.ACCEPTED.code()
+                        && response.getStatusCode() != HttpStatus.NO_CONTENT.code()) {
```

Comment:
I would rather have single expected HttpStatus here. We're testing query cancellation and we do not know what query cancellation REST api is supposed to return?